runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -325,21 +324,32 @@ private static long parseDateTime(String datetime) {
       Iterator<JsonNode> iter = events.elements();
       long timestamp = Long.MAX_VALUE;
       while (iter.hasNext()) {
-        JsonNode occurredAt = iter.next().path(EventField.EventOccurredAt.fieldName());
+        JsonNode event = iter.next();
+        JsonNode occurredAt = event.path(EventField.EventOccurredAt.fieldName());
         try {
+          long occurredAtMillis;
           if (occurredAt.isTextual()) {
-            timestamp = Math.min(parseDateTime(occurredAt.textValue()), timestamp);
+            occurredAtMillis = parseDateTime(occurredAt.textValue());
           } else if (occurredAt.isNumber()) {
-            timestamp = occurredAt.asLong();
+            occurredAtMillis = occurredAt.asLong();
+          } else {
+            continue;
           }
+          if (event.has(EventField.EventLoggedAt.fieldName())) {
+            long loggedAtSeconds = event.path(EventField.EventLoggedAt.fieldName()).asLong();
```

Comment:
asLong() will coerce non-numeric values into a long, which probably isn't what we want. We should check if it's a number before calling this to make sure we don't get garbage.

Reply:
It matches the behavior in streamingplatform (also accepts numeric String). I would like that to remain consistent. We should also try to just reuse the parsing code for these fields instead of repeating it here, but that requires a few visibility changes.